src/inertia.js

```diff
@@ -118,13 +118,13 @@ export default {
     }
   },
 
-  setPage(page, visitId = this.createVisitId(), replace = false, preserveScroll = false) {
+  setPage(page, visitId = this.createVisitId(), replace = false, preserveScroll = false, preserveState = false) {
     this.incrementProgressBar()
     return this.resolveComponent(page.component).then(component => {
       if (visitId === this.visitId) {
         this.version = page.version
         this.setState(page, replace)
-        this.updatePage(component, page.props)
+        this.updatePage(component, page.props, preserveState)
```

Comment:
Wonder if it might be a good idea to abandon positional arguments at this point in case there's ever any more options that start showing up here.

Reply:
Yup, probably. Good call.